### PR TITLE
Cleanup build script.

### PR DESCRIPTION
**Pull request by votdev**
- Use Dockerfile for podman and docker to reduce duplicate code.
- Add missing options in Dockerfile.

Signed-off-by: Volker Theile <vtheile@suse.com>

### CODE REVIEW

File: build/build-container.sh [hunk 1]
```diff
@@ -2,61 +2,31 @@
...
-buildah config --cmd '["--rgw-backend-store", "dbstore", "--debug-rgw", "1"]' \${ctr}
-buildah commit --rm \${ctr} ${IMAGE_NAME}
-EOF
-  buildah unshare sh ${tmpfile}
-  rm -f ${tmpfile}
+  podman)
+    podman build --tag ${IMAGE_NAME} -f ./Dockerfile.build-container ${CEPH_DIR}/build
+    ;;
+  docker)
+    docker build -t localhost/${IMAGE_NAME} -f ./Dockerfile.build-container ${CEPH_DIR}/build
```
Comment (votdev) [inline build/build-container.sh:19]: Can we use `build -t ${IMAGE_NAME}` here as well instead of `localhost/${IMAGE_NAME}`?

Comment (votdev) [inline build/build-container.sh:19]: @giubacc Can we use `build -t ${IMAGE_NAME}` here as well instead of `localhost/${IMAGE_NAME}`?

Comment (giubacc) [reply build/build-container.sh:19]: sure

Comment (giubacc) [reply build/build-container.sh:19]: no, wait, I spoke too early, I remember I added `localhost` to make the tag name consistent with the name's image created with podman. if that is not changed I would leave localhost in the docker tag name.

Comment (votdev) [reply build/build-container.sh:19]: ok, so lets keep the current code.

**Review by giubacc — APPROVED** [review]

**Pull request merged**